tck/TCK_NOTES.adoc

```diff
@@ -0,0 +1,113 @@
+= Named Graphs =
+
+The TCK relies on a number of included named graphs.
+
+
+== Metadata File ==
+
+Each named graph is declared using a JSON file that references various Cypher script files for importing the graph as well as includes some additional statistical information.
+
+This JSON file is called the metadata file.
+
+=== Statistics Generation ===
+
+The following sections contain queries for the original Neo4j implementation of Cypher that have been used to generate the statistical information for some of the named graphs used in the TCK.
+
+These queries use the APOC procedure library for the generation of JSON results.
+
+=== Advices ===
+
+Some of the statistical information is annotated with additional advices (strings) that describes valid inferences that may be drawn from it in the context of how the given named graph is used in the TCK.
+
+Advices must never influence the semantics of query execution.
+
+
+== Property Statistics ==
+
+Property statistics generally describe how many of a certain kind of entities have a given property as well as how many distinct (different) property values are assigned to them.
+
+Property statistics entries use the empty string as a wildcard token (i.e. for describing any property key, or any node with or without labels).
+
+The metadata file currently may contain three types of advice for property statistics entries.
+
+A property statistics entry annotated with the advice `unique` indicates that each possible property value is at most assigned to one entity in the entry's context.
+
+This is already visible from the included statistics.
+
+However the advice additionally indicates that no included scenario will ever violate this constraint (by performing updates to the graph).
```

Comment:
Is the "(by performing updates to the graph)" an _example_ of how in theory something may violate a constraint, or is it the _only_ way to violate the constraint? If the former, I would suggest "(for example, by performing updates to the graph).". If the latter, I would suggest "by performing updates to the graph." (no parentheses).